### PR TITLE
Retry sending webhook on failure

### PR DESCRIPTION
**Pull request by lastpeony**
https://github.com/ant-media/Ant-Media-Server/issues/6119

### CODE REVIEW

File: src/main/java/io/antmedia/AntMediaApplicationAdapter.java [hunk 1]
```diff
@@ -868,45 +868,40 @@ public StringBuilder notifyHook(String url, String id, String action, String str
...
 			try {
-				response = sendPOST(url, variables);
+				sendPOST(url, variables, appSettings.getWebhookRetryCount());
 			} catch (Exception e) {
-				//Make Exception generi
 				logger.error(ExceptionUtils.getStackTrace(e));
 			}
 		}
-		return response;
+		return null;
```
Comment (mekya) [inline src/main/java/io/antmedia/AntMediaApplicationAdapter.java:876]: If you always return null, please change the return type to void.

File: src/main/java/io/antmedia/AppSettings.java [hunk 2]
```diff
@@ -732,6 +732,10 @@ public class AppSettings implements Serializable{
 	 */
 	private static final String SETTINGS_CLUSTER_COMMUNICATION_KEY = "settings.clusterCommunicationKey";
 
+	private static final String SETTINGS_WEBHOOK_RETRY_COUNT = "settings.webHookRetryCount";
```
Comment (mekya) [inline src/main/java/io/antmedia/AppSettings.java:735]: No need these static fields any more. Just only use the exact field name

**Review by mekya — CHANGES_REQUESTED** [review]
Hi @lastpeony,
Thank you for the implementation. 

I just need two changes as I commented to approve the PR. 

FYI

Cheers
Oguz

File: src/main/java/io/antmedia/AppSettings.java (hunk 2 above)
Comment (lastpeony) [reply src/main/java/io/antmedia/AppSettings.java]: done

File: src/main/java/io/antmedia/AntMediaApplicationAdapter.java (hunk 1 above)
Comment (lastpeony) [reply src/main/java/io/antmedia/AntMediaApplicationAdapter.java:876]: This method doesn't always return null; if successful, it returns the response. However, the return value is not utilized in the product code itself; it is only employed for unit testing purposes.

Comment (mekya) [reply src/main/java/io/antmedia/AntMediaApplicationAdapter.java:889]: >This method doesn't always return null; if successful, it returns the response
The read code here always return null. Please correct if I'm wrong. 

>However, the return value is not utilized in the product code itself; it is only employed for unit testing purposes.
If the real code always return null and it's ok, no need to make it return a value for the unit test. 

It creates ambiguity and makes it harder to maintain.

File: src/main/java/io/antmedia/AppSettings.java (hunk 2 above)
Comment (mekya) [reply src/main/java/io/antmedia/AppSettings.java]: Thank you

**Review by mekya — CHANGES_REQUESTED** [review]
Thank you fixing the one issue. For the second one, I've added my comment

File: src/main/java/io/antmedia/AntMediaApplicationAdapter.java (hunk 1 above)
Comment (lastpeony) [reply src/main/java/io/antmedia/AntMediaApplicationAdapter.java:889]: converted sendPOST to void and edited unit test accordingly.
return value of sendPost has never used on real code anyway.
thanks for the review

**Comment by mekya** [comment]
> Enhance webhook reliability by implementing a retry mechanism with delays and attempts. Handling cases where the response is not OK or encounters an IO exception. Resend webhook with a delay of n milliseconds until the specified number of retry attempts is met. Both retry attempt and delay between retry attempts are configureable through app settings.
> 
> Todo after first look by reviewer: Unit test -- done Functional test if possible
> 
> #6119

**Review by mekya — APPROVED** [review]

**Pull request merged**